uint16 pwm
bool enable